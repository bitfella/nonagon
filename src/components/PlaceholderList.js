import React from 'react';
import PropTypes from 'prop-types';
import PlaceholderItem from 'components/PlaceholderItem';

const PlaceholderList = ({ items }) =>
  [...Array(items)].map((e, i) => <PlaceholderItem key={i} />);

PlaceholderList.propTypes = {
  items: PropTypes.number
};

PlaceholderList.defaultProps = {
  items: 3
};

export default PlaceholderList;
