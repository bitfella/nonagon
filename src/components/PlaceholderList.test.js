import React from 'react';
import { shallow } from 'enzyme';
import PlaceholderList from './PlaceholderList';

describe('PlaceholderList', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<PlaceholderList />);

    expect(wrapper).toMatchSnapshot();
  });
});
