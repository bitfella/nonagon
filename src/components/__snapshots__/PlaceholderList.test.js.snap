// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceholderList renders correctly 1`] = `
Array [
  <PlaceholderItem
    key="0"
  />,
  <PlaceholderItem
    key="1"
  />,
  <PlaceholderItem
    key="2"
  />,
]
`;
